app/src/main/java/com/hedvig/app/feature/home/ui/HomeItemDecoration.kt

```diff
@@ -46,6 +53,31 @@ class HomeItemDecoration : RecyclerView.ItemDecoration() {
                 outRect.top = BASE_MARGIN_SEPTUPLE
             }
         }
+
+        if (item is HomeModel.BigText) {
+            val prev = (parent.adapter as? HomeAdapter)?.items?.getOrNull(position - 1) ?: return
+            if (prev is HomeModel.PSA) {
+                outRect.top = BASE_MARGIN_DOUBLE
+            }
+        }
+
+        if (item is HomeModel.PSA) {
+            divider?.let { divider ->
+                outRect.bottom = divider.intrinsicHeight
+            }
+        }
+    }
+
+    override fun onDraw(c: Canvas, parent: RecyclerView, state: RecyclerView.State) {
+        val adapter = (parent.adapter as? HomeAdapter)
+        parent.children.forEach { view ->
+            val position = parent.getChildAdapterPosition(view)
+            val item = adapter?.items?.getOrNull(position) ?: return
+
+            if (item is HomeModel.PSA) {
+                divider?.draw(c)
```

Comment:
If every PSA-item simply has a divider, can it not be added directly into its layout? That would simplify our code quite a bit 🙂

Reply:
I forgot to add `if (prev is HomeModel.PSA && item is HomeModel.PSA)` so it only to be drawn when there are multiple PSAs

Reply:
Sensible!